ext/dom/php_dom.c

```diff
@@ -1369,11 +1369,16 @@ void dom_normalize (xmlNodePtr nodep)
 
 /* {{{ void dom_set_old_ns(xmlDoc *doc, xmlNs *ns) */
 void dom_set_old_ns(xmlDoc *doc, xmlNs *ns) {
-	xmlNs *cur;
-
 	if (doc == NULL)
 		return;
 
+	ZEND_ASSERT(ns->next == NULL);
```

Comment:
I think this still needs fixing though. Because when we put an ns node back we could be putting one back that's already on the list. This code will then create a cycle.
I need to write a test for this and double check this.

Reply:
Note to self: but then the definition wouldn't be on the ->nsDef member! So that means the loop will never be entered and this function will never execute. So this is fine, but we should still assert this via a test.